imap_processing/ialirt/l0/parse_mag.py

```diff
@@ -387,6 +387,73 @@ def apply_gradiometry_correction(
     return mago_corrected, magnitude
 
 
+def interpolate_spherical(
+    sc_inertial_right: np.ndarray,
+    sc_inertial_decline: np.ndarray,
+    sc_spin_phase: np.ndarray,
+    attitude_time: np.ndarray,
+    target_time: float,
+) -> tuple:
+    """
+    Interpolate spherical coordinates.
+
+    Parameters
+    ----------
+    sc_inertial_right : numpy.ndarray
+        Inertial right ascension for 4 packets 0 to 360 degrees, shape (4).
```

Comment:
I think I must be behind on this, but I thought that the magnetic field vectors were taken at a specific snapshot in time and then transmitted down over 4 packets, so we wouldn't want to do any interpolation/combination, but rather just shifting. The values from packet 3 should be moved to packet 1 because that is when that measurement was made? Is there an I-ALIRT MAG algorithm document we should be looking at to review this?